Compiler error format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t'

Hi,
I'm compiling with the latest esphome release for a esp32-c6. 

```
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32-c6-devkitc-1.html
PLATFORM: Espressif 32 (6.8.0) > Espressif ESP32-C6-DevKitC-1
HARDWARE: ESP32C6 160MHz, 512KB RAM, 8MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-builtin, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-espidf @ 3.50300.0 (5.3.0)
 - tool-cmake @ 3.16.4
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - tool-idf @ 1.0.1
 - tool-mconf @ 1.4060000.20190628 (406.0.0)
 - tool-ninja @ 1.9.0
 - tool-riscv32-esp-elf-gdb @ 11.2.0+20220823
 - tool-xtensa-esp-elf-gdb @ 11.2.0+20230208
 - toolchain-esp32ulp @ 2.35.0-20220830
 - toolchain-riscv32-esp @ 13.2.0+20240530
Reading CMake configuration...
Generating assembly for certificate bundle...
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ off, Compatibility ~ soft
Found 0 compatible libraries
Scanning dependencies...
No dependencies
Building in release mode
```
it compiles ok if i disable the debug, but as soon as i add the logger, i get these errors on different parts of the code:

```
_src/esphome/components/cn105/cn105.cpp:74:38: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   74 |         ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %d", timeout);_
```

I attach the full log of the compiler. 

is there an option that need to be set in the compiler to make it work?
thank you


[esp32-c6.log](https://github.com/user-attachments/files/16678502/esp32-c6.log)

hi,
thank you for answering!
all the log coming from the command was included in the file i attached, but here it is (the command I run is "_esphome run file.yaml_"):

```
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32-c6-devkitc-1.html
PLATFORM: Espressif 32 (6.8.0) > Espressif ESP32-C6-DevKitC-1
HARDWARE: ESP32C6 160MHz, 512KB RAM, 8MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-builtin, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-espidf @ 3.50300.0 (5.3.0)
 - tool-cmake @ 3.16.4
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - tool-idf @ 1.0.1
 - tool-mconf @ 1.4060000.20190628 (406.0.0)
 - tool-ninja @ 1.9.0
 - tool-riscv32-esp-elf-gdb @ 11.2.0+20220823
 - tool-xtensa-esp-elf-gdb @ 11.2.0+20230208
 - toolchain-esp32ulp @ 2.35.0-20220830
 - toolchain-riscv32-esp @ 13.2.0+20240530
Reading CMake configuration...
Generating assembly for certificate bundle...
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ off, Compatibility ~ soft
Found 0 compatible libraries
Scanning dependencies...
No dependencies
Building in release mode
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\atc_mithermometer\atc_mithermometer.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\binary_sensor\automation.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\binary_sensor\binary_sensor.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\binary_sensor\filter.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\climate\climate.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\climate\climate_mode.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\climate\climate_traits.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\climateControls.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\cn105.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\componentEntries.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\cycle_management.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\extraComponents.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\heatpumpFunctions.cpp.o
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\hp_readings.cpp.o
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/climateControls.cpp:1:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
src/esphome/components/cn105/climateControls.cpp: In member function 'void CN105Climate::controlDelegate(const esphome::climate::ClimateCall&)':
src/esphome/components/cn105/climateControls.cpp:32:32: warning: implicitly-declared 'constexpr wantedHeatpumpSettings::wantedHeatpumpSettings(const wantedHeatpumpSettings&)' is deprecated [-Wdeprecated-copy]
   32 |     logCheckWantedSettingsMutex(this->wantedSettings);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:195:29: note: because 'wantedHeatpumpSettings' has user-provided 'wantedHeatpumpSettings& wantedHeatpumpSettings::operator=(const wantedHeatpumpSettings&)'
  195 |     wantedHeatpumpSettings& operator=(const wantedHeatpumpSettings& other) {
      |                             ^~~~~~~~
src/esphome/components/cn105/climateControls.cpp:17:57: note:   initializing argument 1 of 'void logCheckWantedSettingsMutex(wantedHeatpumpSettings)'
   17 | void logCheckWantedSettingsMutex(wantedHeatpumpSettings settings) {
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/cn105/climateControls.cpp:83:36: warning: implicitly-declared 'constexpr wantedHeatpumpSettings::wantedHeatpumpSettings(const wantedHeatpumpSettings&)' is deprecated [-Wdeprecated-copy]
   83 |         logCheckWantedSettingsMutex(this->wantedSettings);
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:195:29: note: because 'wantedHeatpumpSettings' has user-provided 'wantedHeatpumpSettings& wantedHeatpumpSettings::operator=(const wantedHeatpumpSettings&)'
  195 |     wantedHeatpumpSettings& operator=(const wantedHeatpumpSettings& other) {
      |                             ^~~~~~~~
src/esphome/components/cn105/climateControls.cpp:17:57: note:   initializing argument 1 of 'void logCheckWantedSettingsMutex(wantedHeatpumpSettings)'
   17 | void logCheckWantedSettingsMutex(wantedHeatpumpSettings settings) {
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/cn105/climateControls.cpp:87:28: warning: implicitly-declared 'constexpr wantedHeatpumpSettings::wantedHeatpumpSettings(const wantedHeatpumpSettings&)' is deprecated [-Wdeprecated-copy]
   87 |         this->debugSettings("control (wantedSettings)", this->wantedSettings);
      |         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:195:29: note: because 'wantedHeatpumpSettings' has user-provided 'wantedHeatpumpSettings& wantedHeatpumpSettings::operator=(const wantedHeatpumpSettings&)'
  195 |     wantedHeatpumpSettings& operator=(const wantedHeatpumpSettings& other) {
      |                             ^~~~~~~~
src/esphome/components/cn105/cn105.h:240:72: note:   initializing argument 2 of 'void CN105Climate::debugSettings(const char*, wantedHeatpumpSettings)'
  240 |     void debugSettings(const char* settingName, wantedHeatpumpSettings settings);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\hp_writings.cpp.o
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
Compiling .pioenvs\clima_bimbi_5\src\esphome\components\cn105\uptime_connection_sensor.cpp.o
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/cn105.cpp:2:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
In file included from src/esphome/components/sensor/sensor.h:3,
                 from src/esphome\components\atc_mithermometer\atc_mithermometer.h:4,
                 from src/esphome.h:3:
src/esphome/components/cn105/cn105.cpp: In member function 'void CN105Climate::set_remote_temp_timeout(uint32_t)':
src/esphome/components/cn105/cn105.cpp:74:38: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   74 |         ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %d", timeout);
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:72:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
   72 | #define ESPHOME_LOG_FORMAT(format) format
      |                                    ^~~~~~
src/esphome/core/log.h:151:28: note: in expansion of macro 'esph_log_i'
  151 | #define ESP_LOGI(tag, ...) esph_log_i(tag, __VA_ARGS__)
      |                            ^~~~~~~~~~
src/esphome/components/cn105/cn105.cpp:74:9: note: in expansion of macro 'ESP_LOGI'
   74 |         ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %d", timeout);
      |         ^~~~~~~~
src/esphome/components/cn105/cn105.cpp:74:74: note: format string is defined here
   74 |         ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %d", timeout);
      |                                                                         ~^
      |                                                                          |
      |                                                                          int
      |                                                                         %ld
src/esphome/components/cn105/cn105.cpp: In member function 'void CN105Climate::set_debounce_delay(uint32_t)':
src/esphome/components/cn105/cn105.cpp:81:34: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   81 |     ESP_LOGI(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to %d", delay);
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:72:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
   72 | #define ESPHOME_LOG_FORMAT(format) format
      |                                    ^~~~~~
src/esphome/core/log.h:151:28: note: in expansion of macro 'esph_log_i'
  151 | #define ESP_LOGI(tag, ...) esph_log_i(tag, __VA_ARGS__)
      |                            ^~~~~~~~~~
src/esphome/components/cn105/cn105.cpp:81:5: note: in expansion of macro 'ESP_LOGI'
   81 |     ESP_LOGI(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to %d", delay);
      |     ^~~~~~~~
src/esphome/components/cn105/cn105.cpp:81:65: note: format string is defined here
   81 |     ESP_LOGI(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to %d", delay);
      |                                                                ~^
      |                                                                 |
      |                                                                 int
      |                                                                %ld
src/esphome/components/cn105/cn105.cpp: In member function 'void CN105Climate::setupUART()':
src/esphome/components/cn105/cn105.cpp:95:19: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   95 |     ESP_LOGI(TAG, "setupUART() with baudrate %d", this->parent_->get_baud_rate());
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:72:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
   72 | #define ESPHOME_LOG_FORMAT(format) format
      |                                    ^~~~~~
src/esphome/core/log.h:151:28: note: in expansion of macro 'esph_log_i'
  151 | #define ESP_LOGI(tag, ...) esph_log_i(tag, __VA_ARGS__)
      |                            ^~~~~~~~~~
src/esphome/components/cn105/cn105.cpp:95:5: note: in expansion of macro 'ESP_LOGI'
   95 |     ESP_LOGI(TAG, "setupUART() with baudrate %d", this->parent_->get_baud_rate());
      |     ^~~~~~~~
src/esphome/components/cn105/cn105.cpp:95:47: note: format string is defined here
   95 |     ESP_LOGI(TAG, "setupUART() with baudrate %d", this->parent_->get_baud_rate());
      |                                              ~^
      |                                               |
      |                                               int
      |                                              %ld
cc1plus.exe: some warnings being treated as errors
*** [.pioenvs\clima_bimbi_5\src\esphome\components\cn105\cn105.cpp.o] Error 1
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/componentEntries.cpp:1:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
In file included from src/esphome/components/sensor/sensor.h:3,
                 from src/esphome\components\atc_mithermometer\atc_mithermometer.h:4,
                 from src/esphome.h:3:
src/esphome/components/cn105/componentEntries.cpp: In member function 'virtual void CN105Climate::setup()':
src/esphome/components/cn105/componentEntries.cpp:29:19: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   29 |     ESP_LOGI(TAG, "remote_temp_timeout is set to %d", this->remote_temp_timeout_);
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:72:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
   72 | #define ESPHOME_LOG_FORMAT(format) format
      |                                    ^~~~~~
src/esphome/core/log.h:151:28: note: in expansion of macro 'esph_log_i'
  151 | #define ESP_LOGI(tag, ...) esph_log_i(tag, __VA_ARGS__)
      |                            ^~~~~~~~~~
src/esphome/components/cn105/componentEntries.cpp:29:5: note: in expansion of macro 'ESP_LOGI'
   29 |     ESP_LOGI(TAG, "remote_temp_timeout is set to %d", this->remote_temp_timeout_);
      |     ^~~~~~~~
src/esphome/components/cn105/componentEntries.cpp:29:51: note: format string is defined here
   29 |     ESP_LOGI(TAG, "remote_temp_timeout is set to %d", this->remote_temp_timeout_);
      |                                                  ~^
      |                                                   |
      |                                                   int
      |                                                  %ld
src/esphome/components/cn105/componentEntries.cpp:30:19: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   30 |     ESP_LOGI(TAG, "debounce_delay is set to %d", this->debounce_delay_);
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:72:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
   72 | #define ESPHOME_LOG_FORMAT(format) format
      |                                    ^~~~~~
src/esphome/core/log.h:151:28: note: in expansion of macro 'esph_log_i'
  151 | #define ESP_LOGI(tag, ...) esph_log_i(tag, __VA_ARGS__)
      |                            ^~~~~~~~~~
src/esphome/components/cn105/componentEntries.cpp:30:5: note: in expansion of macro 'ESP_LOGI'
   30 |     ESP_LOGI(TAG, "debounce_delay is set to %d", this->debounce_delay_);
      |     ^~~~~~~~
src/esphome/components/cn105/componentEntries.cpp:30:46: note: format string is defined here
   30 |     ESP_LOGI(TAG, "debounce_delay is set to %d", this->debounce_delay_);
      |                                             ~^
      |                                              |
      |                                              int
      |                                             %ld
src/esphome/components/cn105/componentEntries.cpp: In member function 'void CN105Climate::set_update_interval(uint32_t)':
src/esphome/components/cn105/componentEntries.cpp:59:19: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   59 |     ESP_LOGD(TAG, "Setting update interval to %d", update_interval);
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:72:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
   72 | #define ESPHOME_LOG_FORMAT(format) format
      |                                    ^~~~~~
src/esphome/core/log.h:152:28: note: in expansion of macro 'esph_log_d'
  152 | #define ESP_LOGD(tag, ...) esph_log_d(tag, __VA_ARGS__)
      |                            ^~~~~~~~~~
src/esphome/components/cn105/componentEntries.cpp:59:5: note: in expansion of macro 'ESP_LOGD'
   59 |     ESP_LOGD(TAG, "Setting update interval to %d", update_interval);
      |     ^~~~~~~~
src/esphome/components/cn105/componentEntries.cpp:59:48: note: format string is defined here
   59 |     ESP_LOGD(TAG, "Setting update interval to %d", update_interval);
      |                                               ~^
      |                                                |
      |                                                int
      |                                               %ld
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/cycle_management.cpp:2:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/extraComponents.cpp:1:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
In file included from src/esphome/components/sensor/sensor.h:3,
                 from src/esphome\components\atc_mithermometer\atc_mithermometer.h:4,
                 from src/esphome.h:3:
src/esphome/components/cn105/cycle_management.cpp: In member function 'void cycleManagement::deferCycle()':
src/esphome/components/cn105/cycle_management.cpp:32:29: error: format '%d' expects argument of type 'int', but argument 5 has type 'uint32_t' {aka 'long unsigned int'} [-Werror=format=]
   32 |     ESP_LOGI(LOG_CYCLE_TAG, "Defering cycle trigger of %d ms", delay);
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/core/log.h:72:36: note: in definition of macro 'ESPHOME_LOG_FORMAT'
   72 | #define ESPHOME_LOG_FORMAT(format) format
      |                                    ^~~~~~
src/esphome/core/log.h:151:28: note: in expansion of macro 'esph_log_i'
  151 | #define ESP_LOGI(tag, ...) esph_log_i(tag, __VA_ARGS__)
      |                            ^~~~~~~~~~
src/esphome/components/cn105/cycle_management.cpp:32:5: note: in expansion of macro 'ESP_LOGI'
   32 |     ESP_LOGI(LOG_CYCLE_TAG, "Defering cycle trigger of %d ms", delay);
      |     ^~~~~~~~
src/esphome/components/cn105/cycle_management.cpp:32:57: note: format string is defined here
   32 |     ESP_LOGI(LOG_CYCLE_TAG, "Defering cycle trigger of %d ms", delay);
      |                                                        ~^
      |                                                         |
      |                                                         int
      |                                                        %ld
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/heatpumpFunctions.cpp:1:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/hp_readings.cpp:1:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
src/esphome/components/cn105/hp_readings.cpp: In member function 'void CN105Climate::getSettingsFromResponsePacket()':
src/esphome/components/cn105/hp_readings.cpp:221:25: warning: implicitly-declared 'constexpr heatpumpSettings::heatpumpSettings(const heatpumpSettings&)' is deprecated [-Wdeprecated-copy]
  221 |     this->heatpumpUpdate(receivedSettings);
      |     ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:146:23: note: because 'heatpumpSettings' has user-provided 'heatpumpSettings& heatpumpSettings::operator=(const heatpumpSettings&)'
  146 |     heatpumpSettings& operator=(const heatpumpSettings& other) {
      |                       ^~~~~~~~
src/esphome/components/cn105/cn105.h:222:42: note:   initializing argument 1 of 'void CN105Climate::heatpumpUpdate(heatpumpSettings)'
  222 |     void heatpumpUpdate(heatpumpSettings settings);
      |                         ~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/cn105/hp_readings.cpp: In member function 'void CN105Climate::heatpumpUpdate(heatpumpSettings)':
src/esphome/components/cn105/hp_readings.cpp:460:24: warning: implicitly-declared 'constexpr heatpumpSettings::heatpumpSettings(const heatpumpSettings&)' is deprecated [-Wdeprecated-copy]
  460 |     this->debugSettings("current", this->currentSettings);
      |     ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:146:23: note: because 'heatpumpSettings' has user-provided 'heatpumpSettings& heatpumpSettings::operator=(const heatpumpSettings&)'
  146 |     heatpumpSettings& operator=(const heatpumpSettings& other) {
      |                       ^~~~~~~~
src/esphome/components/cn105/cn105.h:239:66: note:   initializing argument 2 of 'void CN105Climate::debugSettings(const char*, heatpumpSettings)'
  239 |     void debugSettings(const char* settingName, heatpumpSettings settings);
      |                                                 ~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/cn105/hp_readings.cpp:461:24: warning: implicitly-declared 'constexpr heatpumpSettings::heatpumpSettings(const heatpumpSettings&)' is deprecated [-Wdeprecated-copy]
  461 |     this->debugSettings("received", settings);
      |     ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:146:23: note: because 'heatpumpSettings' has user-provided 'heatpumpSettings& heatpumpSettings::operator=(const heatpumpSettings&)'
  146 |     heatpumpSettings& operator=(const heatpumpSettings& other) {
      |                       ^~~~~~~~
src/esphome/components/cn105/cn105.h:239:66: note:   initializing argument 2 of 'void CN105Climate::debugSettings(const char*, heatpumpSettings)'
  239 |     void debugSettings(const char* settingName, heatpumpSettings settings);
      |                                                 ~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/cn105/hp_readings.cpp:462:24: warning: implicitly-declared 'constexpr wantedHeatpumpSettings::wantedHeatpumpSettings(const wantedHeatpumpSettings&)' is deprecated [-Wdeprecated-copy]
  462 |     this->debugSettings("wanted", this->wantedSettings);
      |     ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:195:29: note: because 'wantedHeatpumpSettings' has user-provided 'wantedHeatpumpSettings& wantedHeatpumpSettings::operator=(const wantedHeatpumpSettings&)'
  195 |     wantedHeatpumpSettings& operator=(const wantedHeatpumpSettings& other) {
      |                             ^~~~~~~~
src/esphome/components/cn105/cn105.h:240:72: note:   initializing argument 2 of 'void CN105Climate::debugSettings(const char*, wantedHeatpumpSettings)'
  240 |     void debugSettings(const char* settingName, wantedHeatpumpSettings settings);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/cn105/hp_readings.cpp:468:27: warning: implicitly-declared 'constexpr heatpumpSettings::heatpumpSettings(const heatpumpSettings&)' is deprecated [-Wdeprecated-copy]
  468 |     this->publishStateToHA(settings);
      |     ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
cc1plus.exe: some warnings being treated as errors
src/esphome/components/cn105/Globals.h:146:23: note: because 'heatpumpSettings' has user-provided 'heatpumpSettings& heatpumpSettings::operator=(const heatpumpSettings&)'
  146 |     heatpumpSettings& operator=(const heatpumpSettings& other) {
      |                       ^~~~~~~~
src/esphome/components/cn105/hp_readings.cpp:421:54: note:   initializing argument 1 of 'void CN105Climate::publishStateToHA(heatpumpSettings)'
  421 | void CN105Climate::publishStateToHA(heatpumpSettings settings) {
      |                                     ~~~~~~~~~~~~~~~~~^~~~~~~~
src/esphome/components/cn105/hp_readings.cpp: In member function 'void CN105Climate::checkWideVaneSettings(heatpumpSettings&, bool)':
src/esphome/components/cn105/hp_readings.cpp:511:28: warning: implicitly-declared 'constexpr heatpumpSettings::heatpumpSettings(const heatpumpSettings&)' is deprecated [-Wdeprecated-copy]
  511 |         this->debugSettings("settings", settings);
      |         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:146:23: note: because 'heatpumpSettings' has user-provided 'heatpumpSettings& heatpumpSettings::operator=(const heatpumpSettings&)'
  146 |     heatpumpSettings& operator=(const heatpumpSettings& other) {
      |                       ^~~~~~~~
src/esphome/components/cn105/cn105.h:239:66: note:   initializing argument 2 of 'void CN105Climate::debugSettings(const char*, heatpumpSettings)'
  239 |     void debugSettings(const char* settingName, heatpumpSettings settings);
      |                                                 ~~~~~~~~~~~~~~~~~^~~~~~~~
*** [.pioenvs\clima_bimbi_5\src\esphome\components\cn105\componentEntries.cpp.o] Error 1
cc1plus.exe: some warnings being treated as errors
*** [.pioenvs\clima_bimbi_5\src\esphome\components\cn105\cycle_management.cpp.o] Error 1
In file included from src/esphome.h:28,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/hp_writings.cpp:1:
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_drop_mode' [-Wmissing-field-initializers]
   62 | };
      | ^
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::cca_low_tx_pwr' [-Wmissing-field-initializers]
src/esphome\components\esp32_ble\const_esp32c6.h:62:1: warning: missing initializer for member 'esp_bt_controller_config_t::csa2_select' [-Wmissing-field-initializers]
src/esphome/components/cn105/hp_writings.cpp: In member function 'void CN105Climate::sendWantedSettingsDelegate()':
src/esphome/components/cn105/hp_writings.cpp:246:24: warning: implicitly-declared 'constexpr wantedHeatpumpSettings::wantedHeatpumpSettings(const wantedHeatpumpSettings&)' is deprecated [-Wdeprecated-copy]
  246 |     this->debugSettings("wantedSettings", wantedSettings);
      |     ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/esphome/components/cn105/Globals.h:195:29: note: because 'wantedHeatpumpSettings' has user-provided 'wantedHeatpumpSettings& wantedHeatpumpSettings::operator=(const wantedHeatpumpSettings&)'
  195 |     wantedHeatpumpSettings& operator=(const wantedHeatpumpSettings& other) {
      |                             ^~~~~~~~
src/esphome/components/cn105/cn105.h:240:72: note:   initializing argument 2 of 'void CN105Climate::debugSettings(const char*, wantedHeatpumpSettings)'
  240 |     void debugSettings(const char* settingName, wantedHeatpumpSettings settings);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
========================================================= [FAILED] Took 56.49 seconds =========================================================
```

In order to use the latest versione of espidf i just included it i the yaml file, here is it:

```
substitutions:
  name: clima_bimbi_5
  friendly_name: Clima Bambini 5

esphome:
  name: ${name}
  platformio_options:
    board: esp32-c6-devkitc-1
    board_build.core: esp32
    board_build.f_cpu: 160000000L
    board_build.flash_mode: qio
    board_build.mcu: esp32c6
    board_upload.maximum_ram_size: 524288
    board_upload.maximum_size: 8388608
    board_upload.speed: 460800
    
esp32:
  # ESP32-C6-DevKitC-1-N8
  board: esp32-c6-devkitc-1
  flash_size: 8MB
  partitions: config\partitions_zb3.csv
  framework:
    type: esp-idf
    version: 5.3.0
    platform_version: 6.8.0
    sdkconfig_options:
      CONFIG_ESPTOOLPY_FLASHSIZE_8MB: y
      CONFIG_BT_BLE_50_FEATURES_SUPPORTED: y
      CONFIG_BT_BLE_42_FEATURES_SUPPORTED: y
  variant: esp32c6

external_components:
  - source: github://echavet/MitsubishiCN105ESPHome
  
  
wifi:
  id: clima_wifi
  ssid: xxxxxx-xxxxxxx
  password: xxxxxxx

esp32_ble_tracker:
  id: ble_tracker
  scan_parameters:
    active: false
    continuous: false
    
text_sensor:
  - platform: ble_scanner
    name: "BLE Devices Scanner"

sensor:
  - platform: atc_mithermometer
    id: bambini_sensore_ble
    mac_address: A4:C1:38:07:CB:DF
    temperature:
      name: "ATC Temperature"
    humidity:
      name: "ATC Humidity"
    battery_level:
      name: "ATC Battery-Level"
    battery_voltage:
      name: "ATC Battery-Voltage"
    signal_strength:
      name: "ATC Signal"
    
uart:
  id: CLIMA_UART
  baud_rate: 2400
  tx_pin: GPIO10
 rx_pin: GPIO11
  
# Climate entity configuration
climate:
  - platform: cn105
    name: "Clima Bimbi"
    update_interval: 4s        # update interval can be ajusted after a first run and logs monitoring* 
    debounce_delay : 500ms
    
logger:
  level: DEBUG
  hardware_uart: USB_SERIAL_JTAG
  
ota:
  - platform: esphome
```

--- a/components/cn105/Globals.h
+++ b/components/cn105/Globals.h
@@ -2,6 +2,8 @@
 #include <esphome.h>
 #include "esphome/components/uart/uart.h"
 
+using namespace esphome;
+
 #if defined(ARDUINO) && ARDUINO >= 100
 #include "Arduino.h"
 #endif

--- a/components/cn105/cn105.cpp
+++ b/components/cn105/cn105.cpp
@@ -71,14 +71,17 @@ void CN105Climate::set_remote_temp_timeout(uint32_t timeout) {
     if (timeout == 4294967295) {
         ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to never.");
     } else {
-        ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %lu", timeout);
+        //ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %lu", timeout);        
+        log_info_uint32(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to ", timeout);
+
         this->pingExternalTemperature();
     }
 }
 
 void CN105Climate::set_debounce_delay(uint32_t delay) {
     this->debounce_delay_ = delay;
-    ESP_LOGI(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to %lu", delay);
+    //ESP_LOGI(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to %lu", delay);
+    log_info_uint32(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to ", delay);
 }
 
 int CN105Climate::get_compressor_frequency() {
@@ -92,8 +95,8 @@ bool CN105Climate::is_operating() {
 // SERIAL_8E1
 void CN105Climate::setupUART() {
 
-    ESP_LOGI(TAG, "setupUART() with baudrate %lu", this->parent_->get_baud_rate());
-
+    //ESP_LOGI(TAG, "setupUART() with baudrate %lu", this->parent_->get_baud_rate());
+    log_info_uint32(TAG, "setupUART() with baudrate ", this->parent_->get_baud_rate());
     this->setHeatpumpConnected(false);
     this->isUARTConnected_ = false;
 

--- a/components/cn105/cn105.cpp
+++ b/components/cn105/cn105.cpp
@@ -71,14 +71,14 @@ void CN105Climate::set_remote_temp_timeout(uint32_t timeout) {
     if (timeout == 4294967295) {
         ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to never.");
     } else {
-        ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %d", timeout);
+        ESP_LOGI(LOG_ACTION_EVT_TAG, "set_remote_temp_timeout is set to %lu", timeout);
         this->pingExternalTemperature();
     }
 }
 
 void CN105Climate::set_debounce_delay(uint32_t delay) {
     this->debounce_delay_ = delay;
-    ESP_LOGI(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to %d", delay);
+    ESP_LOGI(LOG_ACTION_EVT_TAG, "set_debounce_delay is set to %lu", delay);
 }
 
 int CN105Climate::get_compressor_frequency() {
@@ -92,7 +92,7 @@ bool CN105Climate::is_operating() {
 // SERIAL_8E1
 void CN105Climate::setupUART() {
 
-    ESP_LOGI(TAG, "setupUART() with baudrate %d", this->parent_->get_baud_rate());
+    ESP_LOGI(TAG, "setupUART() with baudrate %lu", this->parent_->get_baud_rate());
 
     this->setHeatpumpConnected(false);
     this->isUARTConnected_ = false;

--- a/components/cn105/cn105.h
+++ b/components/cn105/cn105.h
@@ -18,7 +18,9 @@
 
 using namespace esphome;
 
-//class VaneOrientationSelect;  // Déclaration anticipée, définie dans extraComponents
+
+void log_info_uint32(const char* tag, const char* msg, uint32_t value, const char* suffix = "");
+void log_debug_uint32(const char* tag, const char* msg, uint32_t value, const char* suffix = "");
 
 class CN105Climate : public climate::Climate, public Component, public uart::UARTDevice {
 

--- a/components/cn105/componentEntries.cpp
+++ b/components/cn105/componentEntries.cpp
@@ -26,8 +26,10 @@ void CN105Climate::setup() {
     this->nbHeatpumpConnections_ = 0;
 
     ESP_LOGI(TAG, "tx_pin: %d rx_pin: %d", this->tx_pin_, this->rx_pin_);
-    ESP_LOGI(TAG, "remote_temp_timeout is set to %lu", this->remote_temp_timeout_);
-    ESP_LOGI(TAG, "debounce_delay is set to %lu", this->debounce_delay_);
+    //ESP_LOGI(TAG, "remote_temp_timeout is set to %lu", this->remote_temp_timeout_);
+    log_info_uint32(TAG, "remote_temp_timeout is set to ", this->remote_temp_timeout_);
+    //ESP_LOGI(TAG, "debounce_delay is set to %lu", this->debounce_delay_);
+    log_info_uint32(TAG, "debounce_delay is set to ", this->debounce_delay_);
 
     this->setupUART();
     this->sendFirstConnectionPacket();
@@ -56,7 +58,9 @@ void CN105Climate::loop() {
 
 uint32_t CN105Climate::get_update_interval() const { return this->update_interval_; }
 void CN105Climate::set_update_interval(uint32_t update_interval) {
-    ESP_LOGD(TAG, "Setting update interval to %lu", update_interval);
+    //ESP_LOGD(TAG, "Setting update interval to %lu", update_interval);
+    log_debug_uint32(TAG, "Setting update interval to ", update_interval);
+
     this->update_interval_ = update_interval;
     this->autoUpdate = (update_interval != 0);
 }

--- a/components/cn105/componentEntries.cpp
+++ b/components/cn105/componentEntries.cpp
@@ -26,8 +26,8 @@ void CN105Climate::setup() {
     this->nbHeatpumpConnections_ = 0;
 
     ESP_LOGI(TAG, "tx_pin: %d rx_pin: %d", this->tx_pin_, this->rx_pin_);
-    ESP_LOGI(TAG, "remote_temp_timeout is set to %d", this->remote_temp_timeout_);
-    ESP_LOGI(TAG, "debounce_delay is set to %d", this->debounce_delay_);
+    ESP_LOGI(TAG, "remote_temp_timeout is set to %lu", this->remote_temp_timeout_);
+    ESP_LOGI(TAG, "debounce_delay is set to %lu", this->debounce_delay_);
 
     this->setupUART();
     this->sendFirstConnectionPacket();
@@ -39,7 +39,7 @@ void CN105Climate::setup() {
  * This function is called repeatedly in the main program loop.
  */
 void CN105Climate::loop() {
-    if (!this->processInput()) {                                            // if we don't get an input: no read op
+    if (!this->processInput()) {                                            // if we don't get any input: no read op
         if ((this->wantedSettings.hasChanged) && (!this->loopCycle.isCycleRunning())) {
             this->checkPendingWantedSettings();
         } else {
@@ -56,7 +56,7 @@ void CN105Climate::loop() {
 
 uint32_t CN105Climate::get_update_interval() const { return this->update_interval_; }
 void CN105Climate::set_update_interval(uint32_t update_interval) {
-    ESP_LOGD(TAG, "Setting update interval to %d", update_interval);
+    ESP_LOGD(TAG, "Setting update interval to %lu", update_interval);
     this->update_interval_ = update_interval;
     this->autoUpdate = (update_interval != 0);
 }

--- a/components/cn105/cycle_management.cpp
+++ b/components/cn105/cycle_management.cpp
@@ -29,7 +29,7 @@ void cycleManagement::deferCycle() {
     uint32_t delay = DEFER_SCHEDULE_UPDATE_LOOP_DELAY;
 #endif
 
-    ESP_LOGI(LOG_CYCLE_TAG, "Defering cycle trigger of %d ms", delay);
+    ESP_LOGI(LOG_CYCLE_TAG, "Defering cycle trigger of %lu ms", delay);
     // forces the lastCompleteCycle offset of delay ms to allow a longer rest time
     lastCompleteCycleMs = CUSTOM_MILLIS + delay;
 

--- a/components/cn105/cycle_management.cpp
+++ b/components/cn105/cycle_management.cpp
@@ -29,7 +29,8 @@ void cycleManagement::deferCycle() {
     uint32_t delay = DEFER_SCHEDULE_UPDATE_LOOP_DELAY;
 #endif
 
-    ESP_LOGI(LOG_CYCLE_TAG, "Defering cycle trigger of %lu ms", delay);
+    //ESP_LOGI(LOG_CYCLE_TAG, "Defering cycle trigger of %lu ms", delay);
+    log_info_uint32(LOG_CYCLE_TAG, "Defering cycle trigger of  ", delay, " ms");
     // forces the lastCompleteCycle offset of delay ms to allow a longer rest time
     lastCompleteCycleMs = CUSTOM_MILLIS + delay;
 

--- a/components/cn105/utils.cpp
+++ b/components/cn105/utils.cpp
@@ -1,7 +1,20 @@
 #include "cn105.h"
 #include "Globals.h"
 
-
+void log_info_uint32(const char* tag, const char* msg, uint32_t value, const char* suffix) {
+#if __GNUC__ >= 11
+    ESP_LOGI(tag, "%s %lu %s", msg, (unsigned long)value, suffix);
+#else
+    ESP_LOGI(tag, "%s %u %s", msg, (unsigned int)value, suffix);
+#endif
+}
+void log_debug_uint32(const char* tag, const char* msg, uint32_t value, const char* suffix) {
+#if __GNUC__ >= 11
+    ESP_LOGD(tag, "%s %lu %s", msg, (unsigned long)value, suffix);
+#else
+    ESP_LOGD(tag, "%s %u %s", msg, (unsigned int)value, suffix);
+#endif
+}
 
 bool CN105Climate::hasChanged(const char* before, const char* now, const char* field, bool checkNotNull) {
     if (now == NULL) {
@@ -312,4 +325,5 @@ void CN105Climate::testMutex() {
     this->testCase1();
 
 }
+
 #endif

--- a/esp32-test.yaml
+++ b/esp32-test.yaml
@@ -1,5 +1,5 @@
 substitutions:
-  name: "esp32-hp"
+  name: "esp32-hp-test"
   friendly_name: Clim Sejour
 
 esphome:
@@ -10,6 +10,8 @@ esp32:
   board: esp32doit-devkit-v1
   framework:
     type: esp-idf     
+    #version: 5.3.0
+    #platform_version: 6.8.0
     #type: arduino
 
 
@@ -80,7 +82,8 @@ api:
     key: !secret encryption_key_sejour
 
 ota:  
-  password: !secret ota_pwd
+  - platform: esphome
+    password: !secret ota_pwd
 
 wifi:
   #ssid: !secret wifi_ssid
@@ -110,10 +113,10 @@ time:
 
 # this setting is for local component (code is not on github)
 external_components:
-  - source: github://echavet/MitsubishiCN105ESPHome
-#  - source:
-#      type: local
-#      path: components 
+#   - source: github://echavet/MitsubishiCN105ESPHome
+ - source:
+     type: local
+     path: components 
 
         # Text sensors with general information.
 text_sensor:

--- a/esp32-test.yaml
+++ b/esp32-test.yaml
@@ -12,6 +12,7 @@ esp32:
     type: esp-idf     
     #version: 5.3.0
     #platform_version: 6.8.0
+
     #type: arduino
 
 

--- a/hp-sejour.yaml
+++ b/hp-sejour.yaml
@@ -82,7 +82,7 @@ api:
 
 ota:  
   password: !secret ota_pwd
-
+  platform: esphome
 wifi:
   #ssid: !secret wifi_ssid
   #password: !secret wifi_password

